### DISCUSSION TITLE
September Monthly Challenge

### DISCUSSION THREAD

**Discussion by BekahHW** [Virtual-Coffee/VC-Community-Docs#185] [Monthly Challenges]
@Virtual-Coffee/monthly-challenge 

So September is [prep-tember](https://dev.to/devteam/preptember-start-getting-ready-for-hacktoberfest-this-month-58cn) for Hacktoberfest. 

1. Is there a way, we can make prep-tember a monthly challenge that everyone can participate in? We would really love to get some maintainers who can commit to VC-Hacktoberfest Initiative (just meaning they'll prepare their repos to be clear and welcoming).
         a. Maybe we expand this to everyone. Maintainers work on their repos. Mentors sign up and create clear office hours, specify what they can mentor, how many ppl they can mentor, and maybe contribute to docs on how to be a good contributor? And contributors familiarize themselves with the process and also contribute to the contributors docs?
        b. So maybe this becomes a whole community challenge to create a comprehensive guide to hacktoberfest? If we do it in a repo (maybe the existing [open source](https://github.com/Virtual-Coffee/open-source) one, we can aim to have 3 commits per participant in the challenge, and for maintainers to update their repos and report back?
2. Is anyone interested in being the leader for this challenge

**Comment by Andrew-Bush** [comment]
Thank you for linking the DEV article.  That gives me some much needed context.  I didn't have much of an online developer community presence until spring of this year, so I have never done this before.

1. I thnk there can be lots of ways to promote active participation.  I learned from myself and others during Demo In Public that people find motivation in deadlines or accountability to others.  No one wants to show up at the weekly check-in to say they didn't do anything.  
    - So, I see 3 primary ways to Participate. Mentor, Contribute, Maintain.  Folks can pick one and specialize or they can do any combination of them all.  I think the important thing is to create clear paths to participation.
    - I didn't know we had an `open source` repo.  This is great!  3 commits may be a really good bar to set.  Again, I think having clear paths to making those commits is important.  Mentors can definitely help with that.
2. You already know that I am interested in leading.  I will need help though, I cannot do this all on my own.

**Reply by aurelieverrot** [reply]
I would love to help with leading as well.

For prep-tember I think we need someone to explain what it is to be a maintainer, how to create good issues, have the read-me prepared, and also what it is to be maintainer during Hacktoberfest (how to manage the flow of people, avoid the bad contributors, ...)

I think we have enough people with experience to explain this in VC.
This can be done the first week.

Then there will be the need to mentor future maintainers to do the work that needs to be done.

Someone to coordinate the switch from prep-tember to Hacktoberfest: having the list of repos, check with the new maintainers the labels, read-mes...
That way we can start helping members to contribute to HF from day 1. 
That I can do, I am ok to help both Sept and Oct.

I am ok to help both months, I can do the coordination

**Comment by dominicduffin1** [comment]
I think the idea of creating a comprehensive guide to Hacktoberfest during the preptember month. I also think it might be a good idea to have maybe a couple of events where we can get together to chat about Hacktoberfest and people who haven't contributed before could ask questions so that we can get everyone who wants to participate comfortable with contributing and participating when October starts. I could help with coordinating these.

I am also happy to help with creating the documentation for Hacktoberfest.

**Comment by tkshill** [comment]
Last year we had a general written guide to open source, but  not one for hacktoberfest specifically. That could be interesting. I'm trying to recall if digitalocean (last year's main org sponsor for hacktoberfest's activities) puts out a guide beyond the blurb on their website. But yes, super down for giving folks more context. Many people never hear of hacktoberfest before we talk about it.

@dominicduffin1 I can show you what docs we currently have.

@aurelieverrot it would be great to revisit (and hopefully expand) our list of VC open source repos we support.

Over the last year, we've had our readmes and other best practices evolve slightly, happy to chat with folks about consolidating.

**Comment by BekahHW** [comment]
We used [this](https://www.digitalocean.com/community/tech_talks/getting-the-most-out-of-open-source) last year as part of our guide to Hacktoberfest. Shoutout to @nickytonline for the presentation!

**Reply by BekahHW** [reply]
@tkshill you had the guide for maintainers, right? I can't find it anywhere.

**Reply by BekahHW** [reply]
Another [good resource](https://dev.to/devteam/preptember-start-getting-ready-for-hacktoberfest-this-month-58cn)

A [great guide](https://hacktoberfest.digitalocean.com/details/) from digital ocean.

More random thoughts. There are lots of great resources out there, so we don't necessarily create something brandnew, unless we think it's more important for it to be VC-made.

Maybe September becomes a checklist to prepare. To complete the monthly challenge, you sign up as a Maintainer, Contributor, or Mentor. Each of these roles has a checklist they need to complete to fulfill the monthly challenge. For example, Maintainers should review old issues and close existing issues that aren't necessary, add new issues that have a checklist of how to make it a well-written issue, has read a good contributing.md file and has a good one in their repo, has a clear readme, has at least one issue with the Hacktoberfest label, etc. (Or maybe we pair maintainers with checklists and they check it off for the other person).

**Reply by danieltott** [reply]
Totally agree with @BekahHW here. I feel like we can probably provide support for maintainers too - maybe even provide a way for (if they want) VC members to like add issues to repos? Like - it's hard work to create issues even if you know they exist, so maybe something non-maintainer members can do for the month is literally create good issues for repos they use? Not solve them, not create PRs, but literally the challenge in September is "write issues"

This idea just popped in my head but I am excited by it...gonna do the dishes now and think about it...

**Reply by danieltott** [reply]
JK dishes can wait

Another thing I was thinking was it might be good to have a very small crew whose job it is to just find good hacktoberfest repos to use in October - like their job is not to fix up repos or do other things but just find and vet repos. It'd be great to have a mix of technologies for instance.

**Reply by BekahHW** [reply]
> 
> Another thing I was thinking was it might be good to have a very small crew whose job it is to just find good hacktoberfest repos to use in October - like their job is not to fix up repos or do other things but just find and vet repos. It'd be great to have a mix of technologies for instance.

@danieltott you're on a roll. These are GREAT!

**Reply by dominicduffin1** [reply]
@danieltott I think writing issues and finding and vetting repositories would be excellent for September's challenge!

**Reply by nickytonline** [reply]
My talk from last year for Digital Ocean's Hacktoberfest event has some good knowledge nuggets in it for maintainers and contributors, if you need another resource. https://www.iamdeveloper.com/pages/talks/#heading-getting-the-most-out-of-open-source

**Comment by aurelieverrot** [comment]
Retro:

![image](https://user-images.githubusercontent.com/50857179/141593705-686a0a32-8c74-4ef9-be5c-f674227384e3.png)
